src/main/java/com/mcmiddleearth/mcmechat/listener/PlayerListener.java

```diff
@@ -42,6 +42,61 @@ public void onJoin(PlayerJoinEvent e) {
+            String msg = joinMsg.getString("join_messages.group_root");
+            e.getPlayer().sendMessage(msg);
+        }else if(e.getPlayer().hasPermission("group.buildvalar")){
+            String msg = joinMsg.getString("join_messages.group_buildvalar");
+            e.getPlayer().sendMessage(msg);
+        }else if(e.getPlayer().hasPermission("group.enforcevalar")){
+            String msg = joinMsg.getString("join_messages.group_enforcevalar");
+            e.getPlayer().sendMessage(msg);
+        }else if(e.getPlayer().hasPermission("group.valar")){
+            String msg = joinMsg.getString("join_messages.group_valar");
+            e.getPlayer().sendMessage(msg);
+        }else if(e.getPlayer().hasPermission("group.headdeveloper")){
+            String msg = joinMsg.getString("join_messages.group_headdeveloper");
+            e.getPlayer().sendMessage(msg);
+        }else if(e.getPlayer().hasPermission("group.headguide")){
+            String msg = joinMsg.getString("join_messages.group_headguide");
+            e.getPlayer().sendMessage(msg);
+        }else if(e.getPlayer().hasPermission("group.designer")){
+            String msg = joinMsg.getString("join_messages.group_designer");
+            e.getPlayer().sendMessage(msg);
+        }else if(e.getPlayer().hasPermission("group.assistant")){
+            String msg = joinMsg.getString("join_messages.group_assistant");
+            e.getPlayer().sendMessage(msg);
+        }else if(e.getPlayer().hasPermission("group.guide")){
+            String msg = joinMsg.getString("join_messages.group_guide");
+            e.getPlayer().sendMessage(msg);
+        }else if(e.getPlayer().hasPermission("group.artist")){
+            String msg = joinMsg.getString("join_messages.group_artist");
+            e.getPlayer().sendMessage(msg);
+        }else if(e.getPlayer().hasPermission("group.foreman")){
+            String msg = joinMsg.getString("join_messages.group_foreman");
+            e.getPlayer().sendMessage(msg);
+        }else if(e.getPlayer().hasPermission("group.badge_moderator")){
+            String msg = joinMsg.getString("join_messages.group_badgemoderator");
+            e.getPlayer().sendMessage(msg);
+        }else if(e.getPlayer().hasPermission("group.default")){
+            String msg = joinMsg.getString("join_messages.group_default");
+            e.getPlayer().sendMessage(msg);
+        }else if(e.getPlayer().hasPermission("group.adventurer")){
+            String msg = joinMsg.getString("join_messages.group_adventurer");
+            e.getPlayer().sendMessage(msg);
+        }else if(e.getPlayer().hasPermission("group.commoner")){
+            String msg = joinMsg.getString("join_messages.group_commoner");
+            e.getPlayer().sendMessage(msg);
+        }else if(e.getPlayer().hasPermission("group.oathbreaker")){
+            String msg = joinMsg.getString("join_messages.group_oathbreaker");
+            e.getPlayer().sendMessage(msg);
+        }

```

Comment:
I would not put all group names in here, would be better to read them from the config file too.
The configuration keys could be equal to the group names like "adventurer" and the plugin would loop over all configuration keys and check if the player has a matching ``"group."+config_key`` permission. In this case get the string list in the configuration section and send all lines in there to the player.